models/dag_node.go

```diff
@@ -0,0 +1,75 @@
+package models
```

Comment:
this is not the right place to put this I think, models only contain business models, DAGNode doesn't look like one

Reply:
I have moved the dag_node file to core package